intel_nn_hal/GnaPreparedModel.cpp

```diff
@@ -1522,16 +1825,38 @@ bool GnaPreparedModel::operationFullyConnected(const Operation& operation) {
         return blob;
     };
 
+    auto validateOperand = [&](uint32_t idx, uint32_t offset) {
+        const auto op = mModel.main.operands[idx];
+        auto len_out = op.location.length;
+        if (len_out == 0) {
+            return false;
+        }
+        return true;
+    };
+
     IRBuilder::BuilderFCLayer::FCParams params;
-    params.input.data = getIRBlobFromOperand(operation.inputs[0], 0);
+
+    if(!validateOperand(operation.inputs[0], 0)){
+        //return false;
+    }
     params.input.lifeTime = getV1_3_OperandLifeTime(operation.inputs[0]);
+    params.input.data = getIRBlobFromOperand(operation.inputs[0], 0);
+
 
+    if(!validateOperand(operation.inputs[1], 1)){
+        //return false;
+    }
     params.weights.data = getIRBlobFromOperand(operation.inputs[1], 1);
     params.weights.lifeTime = getV1_3_OperandLifeTime(operation.inputs[1]);
 
+    if (!validateOperand(operation.inputs[2], 2)){
```

Comment:
@anikulk Do you need this if statement. Currently its not returning false.

Reply:
Done